asttokens/asttokens.py

```diff
@@ -31,7 +32,111 @@
   from .util import AstNode
 
 
-class ASTTokens(object):
+class TextRangeBase(six.with_metaclass(abc.ABCMeta, object)):
+  def __init__(self, source_text, filename):
+    # type: (Any, str) -> None
+    # FIXME: Strictly, the type of source_type is one of the six string types, but hard to specify with mypy given
+    # https://mypy.readthedocs.io/en/stable/common_issues.html#variables-vs-type-aliases
+
+    self._filename = filename
+
+    # Decode source after parsing to let Python 2 handle coding declarations.
+    # (If the encoding was not utf-8 compatible, then even if it parses correctly,
+    # we'll fail with a unicode error here.)
+    source_text = six.ensure_text(source_text)
+
+    self._text = source_text
+    self._line_numbers = LineNumbers(source_text)
+
+  @abc.abstractmethod
+  def get_text_positions(self, node, padded, unmarked=False):
+    # type: (AstNode, bool, bool) -> Tuple[Tuple[int, int], Tuple[int, int]]
+    raise NotImplementedError
+
+  def _get_text_positions_unmarked(self, node, padded):
```

Comment:
This and the following methods on this class are moved verbatim from `ASTTokens`.